wgpu-hal/src/metal/device.rs

```diff
@@ -41,27 +49,29 @@ fn create_stencil_desc(
     face: &wgt::StencilFaceState,
     read_mask: u32,
     write_mask: u32,
-) -> metal::StencilDescriptor {
-    let desc = metal::StencilDescriptor::new();
-    desc.set_stencil_compare_function(conv::map_compare_function(face.compare));
-    desc.set_read_mask(read_mask);
-    desc.set_write_mask(write_mask);
-    desc.set_stencil_failure_operation(conv::map_stencil_op(face.fail_op));
-    desc.set_depth_failure_operation(conv::map_stencil_op(face.depth_fail_op));
-    desc.set_depth_stencil_pass_operation(conv::map_stencil_op(face.pass_op));
+) -> Retained<MTLStencilDescriptor> {
+    let desc = unsafe { MTLStencilDescriptor::new() };
```

Comment:
thought: It's sorta weird that constructing a new one of these is `unsafe`… 🤔 What preconditions could possibly need to be upheld? I say this as an Apple platform programming newbie.

Reply:
Generally unsafe new means something about dropping being unsafe/unsound.

Reply:
Mostly same as in https://github.com/gfx-rs/wgpu/pull/5641#discussion_r2066231610: this `unsafe` will likely disappear some day.

Reply:
I've started marking a bunch of these safe in https://github.com/madsmtm/objc2/pull/773 based on Metal 4 porting. I have yet to go through our engine / the full bindings in general to identify more simple things like constructors and getters/setters, and will be sure to include `MTLStencilDescriptor` at least :tada:

Reply:
The `unsafe` blocks in this PR are smaller scoped now that [`objc2-metal` is marked safe in more places](https://github.com/madsmtm/objc2/pull/792).